engine/server/engine/server/engine_connect_server_service.go

```diff
@@ -75,21 +75,21 @@ func (service *EngineConnectServerService) GetEngineInfo(context.Context, *conne
 func (service *EngineConnectServerService) CreateEnclave(ctx context.Context, connectArgs *connect.Request[kurtosis_engine_rpc_api_bindings.CreateEnclaveArgs]) (*connect.Response[kurtosis_engine_rpc_api_bindings.CreateEnclaveResponse], error) {
 	args := connectArgs.Msg
 	logrus.Debugf("args: %+v", args)
-	apiContainerLogLevel, err := logrus.ParseLevel(args.ApiContainerLogLevel)
+	apiContainerLogLevel, err := logrus.ParseLevel(args.GetApiContainerLogLevel())
 	if err != nil {
 		return nil, stacktrace.Propagate(err, "An error occurred parsing the log level string '%v':", args.ApiContainerLogLevel)
 	}
 
 	isProduction := false
-	if args.Mode == kurtosis_engine_rpc_api_bindings.EnclaveMode_PRODUCTION {
+	if args.GetMode() == kurtosis_engine_rpc_api_bindings.EnclaveMode_PRODUCTION {
 		isProduction = true
 	}
 	enclaveInfo, err := service.enclaveManager.CreateEnclave(
 		ctx,
 		service.imageVersionTag,
-		args.ApiContainerVersionTag,
+		args.GetApiContainerVersionTag(),
```

Comment:
Do we want `GetXXXX` and not just `ApiContainerVersionTag` I can't tell why one is preferred over the other but I don't like the new `&shouldFollowAll` syntax

Reply:
The diff between using the field or the getter is that the getter returns the zero value (which is the default value in many cases) if it is `nil`

![image](https://github.com/kurtosis-tech/kurtosis/assets/29951623/98f442d4-8f84-4dbd-992c-0947b6561005)

Why you don't like `&shouldFollowAll` syntax ? What do you suggest to use instead? 

What we are trying to solve with this modifications is this:
```
In Rust, this is problematic, since there is a clear difference between null value ("") and absent value (Option::None), as seen in our current SDK: https://crates.io/crates/kurtosis-sdk/0.80.7
```

Reply:
Users won't have to use string pointers when using the SDK, it's internal. and only for setting the protobuf arguments, and it's necessary to do it on this way if we want to use optional arguments

Reply:
we don't need to change the docs because all the changes are internal, not changes on the SDK's API